### PR TITLE
Adding xml style documentation for trainers

### PR DESCRIPTION
**Pull request by sfilipi**
Adding xml style documentation for trainers, to improve what's currently on docs.microsoft.com. 

Adresses #388

### CODE REVIEW

**Comment by sfilipi** [comment]
Starting with just sdca, averaged perceptron and lbfgs to give a feeling about the changes involved. 
I'll add the rest of the trainers on a next commit.

File: src/Microsoft.ML.StandardLearners/Standard/Online/AveragedPerceptron.cs [hunk 1]
```diff
@@ -37,6 +37,17 @@ public sealed class AveragedPerceptronTrainer :
         internal const string UserNameValue = "Averaged Perceptron";
         internal const string ShortName = "ap";
         internal const string Summary = "Perceptron is a binary classification algorithm that makes its predictions based on a linear function.";
+        internal const string DetailedSummary = @"Perceptron is a classification algorithm that makes its predictions based on a linear function.
```
Comment (glebuk) [inline src/Microsoft.ML.StandardLearners/Standard/Online/AveragedPerceptron.cs]: >Perceptron [](start = 50, length = 10)

I wish there was some way to author them in markdown files...  I know attributes require constants, but is there some way to do this?  We have done this for Py and R.

Comment (sfilipi) [reply src/Microsoft.ML.StandardLearners/Standard/Online/AveragedPerceptron.cs]: The documentation system will support ingesting md files, in the future, and those will go away. 
Didn't want to duplicate what they are doing, but at the same time can't leave the docs with just one liners.

File: src/Microsoft.ML/CSharpApi.cs [hunk 2]
```diff
@@ -106,18 +106,6 @@ public void Add(Microsoft.ML.Models.AnomalyDetectionEvaluator input, Microsoft.M
                 _jsonNodes.Add(Serialize("Models.AnomalyDetectionEvaluator", input, output));
             }
 
-            public Microsoft.ML.Models.AnomalyPipelineEnsemble.Output Add(Microsoft.ML.Models.AnomalyPipelineEnsemble input)
-            {
-                var output = new Microsoft.ML.Models.AnomalyPipelineEnsemble.Output();
```
Comment (glebuk) [inline src/Microsoft.ML/CSharpApi.cs]: >feature [](start = 39, length = 7)

Is it possible to reduce the column count to 80 or at most 120 chars?  It's hard to read.

File: src/Microsoft.ML/CSharpApi.cs [hunk 3]
```diff
@@ -3642,15 +3642,18 @@ namespace Trainers
     {
 
         /// <summary>
-        /// Perceptron is a classification algorithm that makes its predictions based on a linear function. 
+        /// Perceptron is a classification algorithm that makes its predictions based on a linear function.
         /// I.e., for an instance with feature values f0, f1,..., f_D-1, , the prediction is given by the sign of sigma[0,D-1] ( w_i * f_i), where w_0, w_1,...,w_D-1 are the weights computed by the algorithm.
+        ///
```
Comment (sharwell) [inline src/Microsoft.ML/CSharpApi.cs]: :memo: blank lines are ignored in XML (they all collapse to a single space). If you want paragraphs, each block should start with `<para>` and end with `</para>`.

**Comment by sharwell** [comment]
I left a comment above that auto-collapsed. Adding this comment to make sure it's seen.

**Comment by sfilipi** [comment]
Thanks for the note, i'll update on the next iteration.

**Pull request merged**